lib/media/closed_caption_parser.js

```diff
@@ -53,27 +50,34 @@ shaka.media.IClosedCaptionParser = class {
  *
  * @implements {shaka.media.IClosedCaptionParser}
  * @final
+ * @export
  */
 shaka.media.ClosedCaptionParser = class {
-  /** */
-  constructor(mimeType) {
-    /** @private {!shaka.cea.ICeaParser} */
+  /**
+   * @param {string} streamMimeType
+   * @param {string} ceaMimeType
+   */
+  constructor(streamMimeType, ceaMimeType) {
+    /** @private {!shaka.extern.ICeaParser} */
     this.ceaParser_ = new shaka.cea.DummyCeaParser();
 
-    if (mimeType.toLowerCase().includes('video/mp4')) {
-      // MP4 Parser to extract closed caption packets from H.264/H.265 video.
-      this.ceaParser_ = new shaka.cea.Mp4CeaParser();
-    }
-    if (mimeType.toLowerCase().includes('video/mp2t')) {
-      // TS Parser to extract closed caption packets from H.264 video.
-      this.ceaParser_ = new shaka.cea.TsCeaParser();
+    const parserFactory =
+        shaka.media.ClosedCaptionParser.findParser(streamMimeType);
+    if (parserFactory) {
+      this.ceaParser_ = parserFactory();
     }
 
     /**
      * Decoder for decoding CEA-X08 data from closed caption packets.
-     * @private {!shaka.cea.ICaptionDecoder}
+     * @private {!shaka.extern.ICaptionDecoder}
      */
-    this.ceaDecoder_ = new shaka.cea.CeaDecoder();
+    this.ceaDecoder_ = new shaka.cea.DummyCaptionDecoder();
+
+    const decoderFactory =
+        shaka.media.ClosedCaptionParser.findDecoder(ceaMimeType);
```

Comment:
The current CEA decoder supports 608 and 708 without differentiating mimetypes, why add this? I reviewed the rest of the code and only the CEA-608 mimetype is used.

I think instead of registering by mimetype, we should register an entire CEA decoder.

Reply:
Mainly I wanted to follow MIME type convention used by other plugins. But you're right, will fix it.